Failure cluster [f8507dc5...]: TestReconcilerAPIServerLeaseMultiCombined: panic: component globals of kube already registered

### Failure cluster [f8507dc52d738723d446](https://go.k8s.io/triage#f8507dc52d738723d446)

##### Error text:
```
Failed;
=== RUN   TestReconcilerAPIServerLeaseMultiCombined
panic: component globals of kube already registered

goroutine 74468 [running]:
k8s.io/apimachinery/pkg/util/runtime.Must(...)
	/home/prow/go/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery/pkg/util/runtime/runtime.go:258
k8s.io/kubernetes/cmd/kube-apiserver/app/testing.StartTestServer({0x3d24740, 0xc0064efd40}, 0xc00e4deb80, {0xc0000a5f28, 0x6, 0x6}, 0xc001b24f00)
	/home/prow/go/src/k8s.io/kubernetes/cmd/kube-apiserver/app/testing/testserver.go:208 +0x2fdb
k8s.io/kubernetes/cmd/kube-apiserver/app/testing.StartTestServerOrDie({0x3d26830, 0xc0064efd40}, 0xc00e4deb80, {0xc0066d8f28, 0x6, 0x6}, 0xc001b24f00)
	/home/prow/go/src/k8s.io/kubernetes/cmd/kube-apiserver/app/testing/testserver.go:528 +0xc5
k8s.io/kubernetes/test/integration/controlplane.testReconcilersAPIServerLease.func1(0x0)
	/home/prow/go/src/k8s.io/kubernetes/test/integration/controlplane/kube_apiserver_test.go:450 +0x225
created by k8s.io/kubernetes/test/integration/controlplane.testReconcilersAPIServerLease in goroutine 74467
	/home/prow/go/src/k8s.io/kubernetes/test/integration/controlplane/kube_apiserver_test.go:448 +0xbb
;
```
#### Recent failures:
[10/21/2024, 3:40:05 PM ci-kubernetes-integration-master](https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-integration-master/1848358290711384064)


/kind failing-test
/kind flake
/sig api-machinery

--- a/test/integration/controlplane/kube_apiserver_test.go
+++ b/test/integration/controlplane/kube_apiserver_test.go
@@ -24,7 +24,6 @@ import (
 	"net/http"
 	"reflect"
 	"strings"
-	"sync"
 	"testing"
 	"time"
 
@@ -440,22 +439,16 @@ func testReconcilersAPIServerLease(t *testing.T, leaseCount int, apiServerCount 
 
 	instanceOptions := kubeapiservertesting.NewDefaultTestServerOptions()
 
-	wg := sync.WaitGroup{}
 	// 1. start apiServerCount api servers
 	for i := 0; i < apiServerCount; i++ {
 		// start count api server
-		wg.Add(1)
-		go func(i int) {
-			defer wg.Done()
-			server := kubeapiservertesting.StartTestServerOrDie(t, instanceOptions, []string{
-				"--endpoint-reconciler-type", "master-count",
-				"--advertise-address", fmt.Sprintf("10.0.1.%v", i+1),
-				"--apiserver-count", fmt.Sprintf("%v", apiServerCount),
-			}, etcd)
-			apiServerCountServers[i] = server
-		}(i)
+		server := kubeapiservertesting.StartTestServerOrDie(t, instanceOptions, []string{
+			"--endpoint-reconciler-type", "master-count",
+			"--advertise-address", fmt.Sprintf("10.0.1.%v", i+1),
+			"--apiserver-count", fmt.Sprintf("%v", apiServerCount),
+		}, etcd)
+		apiServerCountServers[i] = server
 	}
-	wg.Wait()
 
 	// 2. verify API Server count servers have registered
 	if err := wait.PollImmediate(3*time.Second, 2*time.Minute, func() (bool, error) {
@@ -476,18 +469,14 @@ func testReconcilersAPIServerLease(t *testing.T, leaseCount int, apiServerCount 
 
 	// 3. start lease api servers
 	for i := 0; i < leaseCount; i++ {
-		wg.Add(1)
-		go func(i int) {
-			defer wg.Done()
-			options := []string{
-				"--endpoint-reconciler-type", "lease",
-				"--advertise-address", fmt.Sprintf("10.0.1.%v", i+10),
-			}
-			server := kubeapiservertesting.StartTestServerOrDie(t, instanceOptions, options, etcd)
-			leaseServers[i] = server
-		}(i)
+		options := []string{
+			"--endpoint-reconciler-type", "lease",
+			"--advertise-address", fmt.Sprintf("10.0.1.%v", i+10),
+		}
+		server := kubeapiservertesting.StartTestServerOrDie(t, instanceOptions, options, etcd)
+		leaseServers[i] = server
 	}
-	wg.Wait()
+
 	defer func() {
 		for i := 0; i < leaseCount; i++ {
 			leaseServers[i].TearDownFn()

